packages/javascript/bh-shared-ui/src/hooks/useAvailableEnvironments/useAvailableEnvironments.tsx

```diff
@@ -40,21 +40,33 @@ export function useAvailableEnvironments<T = Environment[]>(options?: QueryOptio
     });
 }
 
-export const selectEnvironment = (environmentId: Environment['id']): QueryOptions<Environment>['select'] => {
-    return (data) => data.find((domain) => domain.id === environmentId);
-};
-
-export const useEnvironment = (
+export const useSelectedEnvironment = (
     environmentId?: Environment['id'] | null,
     options?: Omit<QueryOptions<Environment>, 'select'>
 ) => {
-    const { environmentId: environmentIdParam } = useEnvironmentParams();
-    const selectedEnvironment = environmentId ?? environmentIdParam;
+    const { environmentId: environmentIdParam, environmentAggregation, setEnvironmentParams } = useEnvironmentParams();
+    const searchedEnvironmentId = environmentId ?? environmentIdParam;
 
-    return useAvailableEnvironments({
-        select: selectEnvironment(selectedEnvironment!),
+    const environmentQuery = useAvailableEnvironments({
+        select: (data) => data.find((domain) => domain.id === searchedEnvironmentId),
         refetchOnWindowFocus: false,
-        enabled: !!selectedEnvironment,
+        enabled: !!searchedEnvironmentId,
         ...options,
     });
+
+    const selectEnvironment = (environmentId: EnvironmentQueryParams['environmentId']) => {
+        setEnvironmentParams({ environmentId, environmentAggregation: null });
+    };
+
+    const selectEnvironmentAggregation = (aggregation: EnvironmentQueryParams['environmentAggregation']) => {
+        setEnvironmentParams({ environmentAggregation: aggregation });
+    };
+
+    return {
+        ...environmentQuery,
+        environment: environmentQuery.data,
+        environmentAggregation,
+        selectEnvironment,
+        selectEnvironmentAggregation,
```

Comment:
I ended up including these properties in the return value from this hook out of convience. It allows us to get the environment as `const { environment } = useSelectedEnvironment()` and create a state switch with redux really easily